### PR TITLE
Add's validation so that at least one account is provided in meshmap form.

### LINKED ISSUE #2689: [Extension] Adding required field to meshmap form
**Current Behavior**
Currently, the `Accounts to connect` portion of the Meshmap sign-up form doesn't have a required field. To be able to give a user permission to access meshmap, it is necessary to fill in one of these fields. 

**Desired Situation**
Add a validation where at least one of these fields is necessary to fill in. 
Screenshot 2022-04-01 at 12 36 50 AM

---
[Optional] **Alternatives**

[Optional] **Additional context**

---
**Contributor Resources**

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)
- See Layer5 site designs in this [Figma project](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). Join the [Layer5 Community](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by SAEb-ai**
Signed-off-by: Md Shahbaz Alam <shahbazalam75508@gmail.com>

**Description**

This PR fixes #2689.

**Notes for Reviewers**

- Add's a validation so that at least one account information is provided in meshmap form before the actual submission.
- Video Illustration:
  

https://user-images.githubusercontent.com/69013647/163672942-55746e9b-ffb5-4f4b-a1ad-47d456d403d2.mp4
- Video Illustration after https://github.com/layer5io/layer5/pull/2719#pullrequestreview-943868147:
 

https://user-images.githubusercontent.com/69013647/166219307-2536f377-997c-42a1-9607-5cf9825bccae.mp4

- Final Video Illustration after https://github.com/layer5io/layer5/pull/2719#discussion_r862737978 was resolved:
  

https://user-images.githubusercontent.com/69013647/167074656-26415751-9beb-4a9f-b2e6-4c3643688dbc.mp4

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by SAEb-ai** [comment]
@warunicorn19 @leecalcote Please take a look. Thanks!!

**Comment by l5io** [comment]
🚀 Preview for commit 1f4fc4d0134325e02a4493ac6499848db9672218 at: https://625ae0f61f553d23f390b9ce--layer5ng.netlify.app

**Review by warunicorn19 — CHANGES_REQUESTED** [review]
Okay I see a small problem with this, on the first try if someone doesn't enter any of the accounts, all the fields of the form disappear, ideally the data of the form should be there with just the error message.

**Comment by debo19** [comment]
@SAEb-ai Let's discuss this in the Websites call today, if you're available?

**Comment by leecalcote** [comment]
@SAEb-ai did we get this one figured out?

**Comment by leecalcote** [comment]
Yo, @SAEb-ai, where are we at on this one?

**Comment by leecalcote** [comment]
@warunicorn19 will you take over this abandoned PR?

**Comment by warunicorn19** [comment]
Yup, I'll take a look @leecalcote

**Comment by SAEb-ai** [comment]
@leecalcote Sorry I was giving my university exams. That's why I was a bit inactive on Github. Now it's over. I wish to address the requested changes further. Can I please do that?

**Comment by leecalcote** [comment]
Thank you, sir.

**Comment by leecalcote** [comment]
@SAEb-ai, yes, if you can complete in the next 12 hours.

**Comment by debo19** [comment]
@SAEb-ai updates on this?

File: src/sections/Meshmap/index.js [hunk 1]
```diff
@@ -101,9 +103,19 @@ const Meshmap = () => {
...
+                        setStepNumber(1);
+                        nextStep();
+                      }
+                      else {
+                        setValidateAccounts(true);
+                        firstname.value=values.firstname;
+                        email.value=values.email;
+                        lastname.value=values.lastname;
+                        occupation.value=values.occupation;
+                        org.value=values.org;
```
Comment (SAEb-ai) [inline src/sections/Meshmap/index.js:117]: Can you please give some suggestions how I can change the role value beacause it's saying undefined when I am changing it in the same manner as above?

**Comment by SAEb-ai** [comment]
@warunicorn19 @leecalcote @debo19 Can you please take a look at my above mentioned https://github.com/layer5io/layer5/pull/2719#pullrequestreview-958915420? Thanks!!

**Comment by l5io** [comment]
🚀 Preview for commit e4df8f8c5f0f350e3ab6c400a56893d67027c924 at: https://626fd14da152cf2dba375f1e--layer5ng.netlify.app

File: src/sections/Meshmap/index.js (hunk 1 above)
Comment (SAEb-ai) [reply src/sections/Meshmap/index.js]: Resolved this by tracking the initialvalue of role.

**Comment by SAEb-ai** [comment]
@warunicorn19 @leecalcote @debo19 I have resolved [this issue](https://github.com/layer5io/layer5/pull/2719#discussion_r862737978) now. Can please take a look over the PR now? Thanks!!

**Comment by l5io** [comment]
🚀 Preview for commit 1744a9497a6edef3e7b64fc4b903a7f2db8d6edc at: https://6274bdac6cb5ef2e59ade776--layer5ng.netlify.app

File: src/sections/Meshmap/index.js [hunk 2]
```diff
@@ -122,6 +142,7 @@ const Meshmap = () => {
                         <p>
                           Choose between Twitter, Google, LinkedIn, and GitHub, provide the username/handle of your user account for your preferred identity provider. Selected beta program participants will receive a free Meshery Cloud account and have full access MeshMap enabled for each of the following user accounts that you provide. Please provide at least one account.
                         </p>
+                        {validateAccounts && <p style={{margin:"0px",color:"red"}}>{errorAccounts}</p>}
```
Comment (Chadha93) [inline src/sections/Meshmap/index.js:145]: @SAEb-ai please use className to avoid inline styling.

**Review by Chadha93 — CHANGES_REQUESTED** [review]
@SAEb-ai Thank you for updating the PR, it works well, just a minor change suggested to update then we are good to merge this.

**Review by warunicorn19 — APPROVED** [review]
Looking good, we can take care of the inline styling later.

**Comment by Nikhil-Ladha** [comment]
@warunicorn19 seems like this is needed for the docker-extension form too. Are you planning to take it up?

**Comment by SAEb-ai** [comment]
I can take that @Nikhil-Ladha.Thanks!!

**Comment by Nikhil-Ladha** [comment]
> I can take that @Nikhil-Ladha .Thanks!!

Please try to send it by today.

**Comment by leecalcote** [comment]
Thanks, @SAEb-ai. Many people interested in the Docker Extension given the announcement at DockerCon.

**Comment by SAEb-ai** [comment]
@leecalcote I am working on it and will try to send a PR in a hour!!

**Comment by SAEb-ai** [comment]
Do I also have to implement the roles section @Nikhil-Ladha @leecalcote?

**Comment by Nikhil-Ladha** [comment]
> Do I also have to implement the roles section @Nikhil-Ladha @leecalcote?

Probably, yes.....but waiting for Lee's confirmation.
Considering the form initially never had the account/roles field in them

**Comment by Nikhil-Ladha** [comment]
Hey @SAEb-ai while testing, can you please comment out the hook and just use console.log?

**Pull request merged**